lib/ansible/modules/database/influxdb/influxdb_user.py

```diff
@@ -185,6 +237,12 @@ def main():
         else:
             user_password = user_password or ''
             create_user(module, client, user_name, user_password, admin)
+            changed = True
+
+        if grants:
+            changed = changed and set_user_grants(module, client, user_name, grants)
```

Comment:
This part is never executed if the user already exist due to https://github.com/ansible/ansible/blob/9534fde0244bb8d1115f16d9ff9332aae574bf69/lib/ansible/modules/database/influxdb/influxdb_user.py#L236

If I remove the above part and change the line 243 to 
```
            if set_user_grants(module, client, user_name, grants):               
                 changed = True 
```

Regards,

Reply:
Forgot to mention, that another part is necessary to remove actual grants if "grants" is not provided to the module
```
         if grants:                                                               
             if set_user_grants(module, client, user_name, grants):               
                 changed = True                                                   
         else:                                                                    
             if set_user_grants(module, client, user_name, []):                                                                                                                                          
                 changed = True     
```

Reply:
Sorry for the delay, I added your modifications. Thanks !